### PR TITLE
Code Quality: Removed Vanara from DriveHelpers

### PR DESCRIPTION
**Pull request by 0x5bfa**
### Resolved / Related Issues

- https://github.com/files-community/Files/issues/15000

### Steps used to test these changes

1. Set a long drive label
2. See it in Files properly shown

### CODE REVIEW

File: src/Files.Core.Storage/Contracts/IDeviceWatcher.cs [hunk 1]
```diff
@@ -20,11 +20,6 @@ internal interface IDeviceWatcher
 		/// </summary>
 		event EventHandler<DeviceEventArgs>? ItemDeleted;
 
-		/// <summary>
-		/// Gets invoked when an item changing is detected by the watcher
-		/// </summary>
-		event EventHandler<DeviceEventArgs>? ItemChanged;
```
Comment (yaira2) [inline src/Files.Core.Storage/Contracts/IDeviceWatcher.cs:26]: Was this used for anything?

File: src/Files.App.Storage/Watchers/DeviceWatcher.cs [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+		private Windows.Devices.Enumeration.DeviceWatcher? _winDeviceWatcher;
+
+		private ManagementEventWatcher? _mmiInsertWatcher, _mmiRemoveWatcher, _mmiModifyWatcher;
+
+		/// <inheritdoc/>
+		public bool CanBeStarted
+			=> _winDeviceWatcher?.Status is DeviceWatcherStatus.Created or DeviceWatcherStatus.Stopped or DeviceWatcherStatus.Aborted;
+
+		/// <inheritdoc/>
+		public event EventHandler<DeviceEventArgs>? ItemAdded;
```
Comment (yaira2) [inline src/Files.App.Storage/Watchers/DeviceWatcher.cs:22]: What do item refer to?

File: src/Files.App/Files.App.csproj
Comment (yaira2) [inline src/Files.App/Files.App.csproj:1]: Please revert the changes

File: src/Files.Core.Storage/Contracts/IDeviceWatcher.cs (hunk 1 above)
Comment (0x5bfa) [reply src/Files.Core.Storage/Contracts/IDeviceWatcher.cs:26]: I can't remember why added but I assume there was the same in WindowsDeviceWatcher class, which wraps 'ejected' and 'inserted' events of DeviceManager to 'changed' event.
But I think we shouldn't do that in the watcher and better do that in the service, taking which event it wants.

File: src/Files.App.Storage/Watchers/DeviceWatcher.cs (hunk 2 above)
Comment (0x5bfa) [reply src/Files.App.Storage/Watchers/DeviceWatcher.cs:22]: The device, in this context, the drive.
Do you think it should be simply 'Added'? Other watchers have 'Item' prefixes so I'd like to rename all after this PR in another PR.

Comment (yaira2) [reply src/Files.App.Storage/Watchers/DeviceWatcher.cs:22]: It should be `DeviceAdded`

File: src/Files.Core.Storage/Contracts/IDeviceWatcher.cs (hunk 1 above)
Comment (0x5bfa) [reply src/Files.Core.Storage/Contracts/IDeviceWatcher.cs:26]: Reverted.

File: src/Files.App.Storage/Watchers/DeviceWatcher.cs (hunk 2 above)
Comment (0x5bfa) [reply src/Files.App.Storage/Watchers/DeviceWatcher.cs]: Done.

File: src/Files.App/Files.App.csproj
Comment (0x5bfa) [reply src/Files.App/Files.App.csproj:1]: Done.

**Comment by 0x5bfa** [comment]
This has Vanara removal and is the phase 1. Phase 2 is to merge DriveHelpers into the service but I probably won't do that for the time being as this doesn't involve with Vanara removal.

https://github.com/files-community/Files/blob/af07aa37b62c8f4bc0ceda362b092921f3456cb9/src/Files.App/Utils/Storage/Helpers/DriveHelpers.cs#L135

**Review by yaira2 — APPROVED** [review]
